mail_tracking/tests/test_mail_tracking.py

```diff
@@ -101,13 +99,29 @@ def test_mail_send(self):
         tracking_email = self.env['mail.tracking.email'].search([
             ('mail_id', '=', mail.id),
         ])
-        self.assertEqual(mail.email_to, tracking_email.recipient)
-        self.assertEqual(mail.email_from, tracking_email.sender)
+        return mail, tracking_email
+
+    def test_mail_send(self):
+        controller = MailTrackingController()
+        db = self.env.cr.dbname
+        image = base64.decodestring(BLANK)
+        mail, tracking = self.mail_send()
+        self.assertEqual(mail.email_to, tracking.recipient)
+        self.assertEqual(mail.email_from, tracking.sender)
         with mock.patch(mock_request) as mock_func:
             mock_func.return_value = type('obj', (object,), self.request)
-            res = controller.mail_tracking_open(db, tracking_email.id)
+            res = controller.mail_tracking_open(db, tracking.id)
             self.assertEqual(image, res.response[0])
 
+    def test_smtp_error(self):
+        with mock.patch(mock_send_email) as mock_func:
+            mock_func.return_value = None
+            mock_func.side_effect = Warning('Test error')
```

Comment:
No need to init the error, side_effect works like return_val so simply `Warning`